### PR TITLE
CART-121 added service for checking user

### PR DESCRIPTION
**Pull request by RomanFedytskyi**

### CODE REVIEW

File: dev/publicDev/admin.employees/config.js [hunk 1]
```diff
@@ -14,23 +14,23 @@
         controller: 'AdminEmployeesController',
         controllerAs: 'vm',
         resolve: {
-          resolver: resolver.get
+          resolver: resolver.get2('see-admin-page')
```
Comment (dmitrij-borchuk) [inline dev/publicDev/admin.employees/config.js]: Rename

File: dev/publicDev/admin.employees/config.js [hunk 2]
```diff
@@ -14,23 +14,23 @@
...
         }
       })
       .when('/admin/employees/edit', {
         templateUrl: 'admin.employees/employeesAdd.html',
         controller: 'userEditController',
         controllerAs: 'vm',
         resolve: {
-          resolver: resolver.get
+          resolver: resolver.get2('see-admin-page')
```
Comment (dmitrij-borchuk) [inline dev/publicDev/admin.employees/config.js]: Rename

File: dev/publicDev/admin.employees/config.js [hunk 3]
```diff
@@ -14,23 +14,23 @@
...
         }
       })
       .when('/admin/employees/edit/:id', {
         templateUrl: 'admin.employees/employeesAdd.html',
         controller: 'userEditController',
         controllerAs: 'vm',
         resolve: {
-          resolver: resolver.get
+          resolver: resolver.get2('see-admin-page')
```
Comment (dmitrij-borchuk) [inline dev/publicDev/admin.employees/config.js]: Rename

File: dev/publicDev/admin.got-towed/config.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ config(['$routeProvider', 'resolverProvider', function($routeProvider, resolverP
       templateUrl: 'admin.got-towed/admin.got-towed.html',
       controller: 'AdminGotTowedController as vm',
       resolve: {
-        resolver: resolver.get
+        resolver: resolver.get2('see-admin-page')
```
Comment (dmitrij-borchuk) [inline dev/publicDev/admin.got-towed/config.js]: Rename

File: dev/publicDev/mainpage/homeConfig,js.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ config(['$routeProvider', 'resolverProvider', function($routeProvider, resolverP
     {
       templateUrl: 'mainpage/home.html',
       resolve: {
-        resolver: resolver.get
+        resolver: resolver.get2()
```
Comment (dmitrij-borchuk) [inline dev/publicDev/mainpage/homeConfig,js.js]: Rename

File: dev/publicDev/managers-page/managers-config.js [hunk 6]
```diff
@@ -19,7 +19,7 @@
     var resolver = resolverProvider.$get();
 
     managersRoute.route.resolve = {
-      resolver: resolver.get
+      resolver: resolver.get2('see-manager-page')
```
Comment (dmitrij-borchuk) [inline dev/publicDev/managers-page/managers-config.js]: Rename

File: dev/publicDev/scripts/resolver.js [hunk 7]
```diff
@@ -3,23 +3,40 @@
...
 
-  app.service('resolver', customPageService);
+  app.service('resolver', service);
 
-  customPageService.$inject = ['$q', 'authService'];
+  service.$inject = ['$q', 'authService', 'UserCheckingService', '$location'];
 
-  function customPageService($q, authService) {
+  function service($q, authService, UserCheckingService, $location) {
+    this.get2 = function (action) {
```
Comment (dmitrij-borchuk) [inline dev/publicDev/scripts/resolver.js]: Rename

File: dev/publicDev/scripts/resolver.js [hunk 8]
```diff
@@ -3,23 +3,40 @@
...
+      return _.bind(function () {
+        return resolve(action);
+      }, this);
+    };
     this.get = function () {
```
Comment (dmitrij-borchuk) [inline dev/publicDev/scripts/resolver.js]: remove

File: dev/publicDev/scripts/userCheckingService.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+'use strict';
```
Comment (dmitrij-borchuk) [inline dev/publicDev/scripts/userCheckingService.js:1]: Chech identation

File: dev/publicDev/scripts/userCheckingService.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+      }
+
+      function checkUser(user, roles, action) {
+        var rolesMap = {};
+        var usersActions = [];
+        roles.forEach(function (roles) {
+          rolesMap[roles.name] = roles;
+        });
+        user.roles.forEach(function (roles) {
+          usersActions = usersActions.concat(rolesMap[roles.name].actions[0].name);
```
Comment (dmitrij-borchuk) [inline dev/publicDev/scripts/userCheckingService.js]: Why only first action? (.actions[0])

File: dev/routing/actions.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const path = require('path');
+const DAL = require('../dal/dal.js');
+const Boom = require('boom');
+const Joi = require('joi');
```
Comment (dmitrij-borchuk) [inline dev/routing/actions.js]: What for?

File: dev/routing/actions.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const path = require('path');
+const DAL = require('../dal/dal.js');
+const Boom = require('boom');
```
Comment (dmitrij-borchuk) [inline dev/routing/actions.js]: What for?

File: dev/routing/actions.js [hunk 13]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const path = require('path');
```
Comment (dmitrij-borchuk) [inline dev/routing/actions.js]: What for?

File: dev/routing/actions.js [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+const Boom = require('boom');
+const Joi = require('joi');
+
+module.exports = function (server) {
+
+  server.route({
+    method: 'GET',
+    path: '/api/actions',
+    handler: function (request, reply) {
+        DAL.users.createUser(request.payload, function (err, docs) {
```
Comment (dmitrij-borchuk) [inline dev/routing/actions.js]: "DAL.users.createUser" ???????

File: dev/routing/actions.js [hunk 15]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const path = require('path');
+const DAL = require('../dal/dal.js');
+const Boom = require('boom');
+const Joi = require('joi');
+
+module.exports = function (server) {
+
+  server.route({
```
Comment (dmitrij-borchuk) [inline dev/routing/actions.js]: What for this route?

File: dev/server.js [hunk 16]
```diff
@@ -68,7 +68,7 @@ let configureServer = (server) => {
 
 function startServer() {
   const server = new Hapi.Server();
-  server.connection( {port: 80, routes: {cors: {origin: ['*'],credentials : true}}} );
+  server.connection( {port: 8080, routes: {cors: {origin: ['*'],credentials : true}}} );
```
Comment (dmitrij-borchuk) [inline dev/server.js]: What for?

**Review by dmitrij-borchuk — CHANGES_REQUESTED** [review]

File: dev/publicDev/admin.employees/config.js (hunk 1 above)
Comment (RomanFedytskyi) [reply dev/publicDev/admin.employees/config.js]: fixed

File: dev/publicDev/admin.employees/config.js (hunk 2 above)
Comment (RomanFedytskyi) [reply dev/publicDev/admin.employees/config.js]: fixed

File: dev/publicDev/admin.employees/config.js (hunk 3 above)
Comment (RomanFedytskyi) [reply dev/publicDev/admin.employees/config.js]: fixed

File: dev/publicDev/admin.got-towed/config.js (hunk 4 above)
Comment (RomanFedytskyi) [reply dev/publicDev/admin.got-towed/config.js]: fixed

File: dev/publicDev/mainpage/homeConfig,js.js (hunk 5 above)
Comment (RomanFedytskyi) [reply dev/publicDev/mainpage/homeConfig,js.js]: fixed

File: dev/publicDev/scripts/resolver.js (hunk 7 above)
Comment (RomanFedytskyi) [reply dev/publicDev/scripts/resolver.js]: fixed

File: dev/publicDev/scripts/resolver.js (hunk 8 above)
Comment (RomanFedytskyi) [reply dev/publicDev/scripts/resolver.js]: fixed

File: dev/publicDev/scripts/userCheckingService.js (hunk 10 above)
Comment (RomanFedytskyi) [reply dev/publicDev/scripts/userCheckingService.js]: fixed

File: dev/routing/actions.js (hunk 12 above)
Comment (RomanFedytskyi) [reply dev/routing/actions.js]: fixed

File: dev/routing/actions.js (hunk 15 above)
Comment (RomanFedytskyi) [reply dev/routing/actions.js]: fixed

File: dev/server.js (hunk 16 above)
Comment (RomanFedytskyi) [reply dev/server.js]: fixed

File: dev/publicDev/scripts/userCheckingService.js [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+      return {
+        checkUser: checkUser
+      }
+
+      function checkUser(user, action) {
+        // var length = user.roles.actions.length;
+        // var i = 0;
+        // for (; i < length; i++) {
+        //   user.roles.actions[i] === action ? return true : return false;
+        // }
```
Comment (vetalcore) [inline dev/publicDev/scripts/userCheckingService.js]: remove comments or  write some explanation (TODO)

File: dev/publicDev/scripts/userCheckingService.js [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+            answer = true;
+          } else {
+            if(user.roles === 'user' && action === 'see-user-page') {
+            answer = true;
+          }
+        }
+        return answer;
+      }
+    }
+})();
```
Comment (vetalcore) [inline dev/publicDev/scripts/userCheckingService.js]: dont use angular directly in IIFE block
should be smthing like:
(function(NG, window, whatever){
    //body
})(angular, window, whatever)

File: dev/publicDev/scripts/userCheckingService.js [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+        // for (; i < length; i++) {
+        //   user.roles.actions[i] === action ? return true : return false;
+        // }
+        (user.roles === manager && action === 'see-manager-page') ? return true : return false;
+        (user.roles === admin && action === 'see-admin-page') ? return true : return false;
+        
+        var answer = false;
+        if (user.roles === 'admin' && action === 'see-admin-page') {
+          answer = true;
+        } else {
```
Comment (vetalcore) [inline dev/publicDev/scripts/userCheckingService.js]: rewrite using **} else if (condition) {** instead of  **} else { if (condition)**

**Review by vetalcore — CHANGES_REQUESTED** [review]

File: dev/publicDev/admin.parkingProblem/config.js [hunk 20]
```diff
@@ -3,13 +3,17 @@
...
+  var resolver = resolverProvider.$get();
   $routeProvider.when(
     '/admin/parking_problem',
     {
       templateUrl: 'admin.parkingProblem/admin.parkingProblem.html',
       controller: 'AdminParkingProblemController',
-      controllerAs: 'vm'
+      controllerAs: 'vm',
+      resolve: {
+        resolver: resolver.get('see-admin-page')
```
Comment (vetalcore) [inline dev/publicDev/admin.parkingProblem/config.js]: 'see-admin-page' -> move to some var

File: dev/publicDev/admin.properties/config.js [hunk 21]
```diff
@@ -14,15 +14,15 @@
         controller: 'adminPropertiesCtrl',
         controllerAs: 'vm',
         resolve: {
-          resolver: resolver.get
+          resolver: resolver.get('see-admin-page')
```
Comment (vetalcore) [inline dev/publicDev/admin.properties/config.js]: 'see-admin-page' -> move to some var

File: dev/publicDev/admin.properties/config.js [hunk 22]
```diff
@@ -14,15 +14,15 @@
...
         }
       })
       .when('/admin/properties/add', {
         templateUrl: 'admin.properties/propertyAdd.html',
         controller: 'adminPropertiesCtrl',
         controllerAs: 'vm',
         resolve: {
-          resolver: resolver.get
+          resolver: resolver.get('see-admin-page')
```
Comment (vetalcore) [inline dev/publicDev/admin.properties/config.js]: 'see-admin-page' -> move to some var

File: dev/publicDev/scripts/resolver.js [hunk 23]
```diff
@@ -3,23 +3,35 @@
...
+              authService.setUser(res[0].data);
+
+              if (action && !UserCheckingService.checkUser(user, roles, action)) {
+                $location.path('/');
+              }
+
+              resolve();
+            },
+            function (errRes) {
+              resolve();
```
Comment (vetalcore) [inline dev/publicDev/scripts/resolver.js:31]: resolve(); -> replace with promise's reject method

File: dev/publicDev/scripts/resolver.js [hunk 24]
```diff
@@ -3,23 +3,35 @@
 (function() {
```
Comment (vetalcore) [inline dev/publicDev/scripts/resolver.js]: pass angular as a props

**Review by vetalcore — COMMENTED** [review]
wrap everything onto IIFE blocks to eliminate global scope polutting
dont use angular directly in IIFE blocks (pass it throught props)

File: dev/publicDev/scripts/resolver.js (hunk 23 above)
Comment (dmitrij-borchuk) [reply dev/publicDev/scripts/resolver.js:31]: If user haven't loginned yet, we reseive the error as response on 'currentUser' reques.
This is resolver, it should be resolved even if error appears.

File: dev/publicDev/managers-page/managers-config.js (hunk 6 above)
Comment (RomanFedytskyi) [reply dev/publicDev/managers-page/managers-config.js]: fixed

File: dev/publicDev/scripts/userCheckingService.js (hunk 9 above)
Comment (RomanFedytskyi) [reply dev/publicDev/scripts/userCheckingService.js:1]: fixed

File: dev/routing/actions.js (hunk 13 above)
Comment (RomanFedytskyi) [reply dev/routing/actions.js]: fixed

File: dev/routing/actions.js (hunk 11 above)
Comment (RomanFedytskyi) [reply dev/routing/actions.js]: fixed

File: dev/routing/actions.js (hunk 14 above)
Comment (RomanFedytskyi) [reply dev/routing/actions.js]: fixed

File: dev/publicDev/admin.parkingProblem/config.js (hunk 20 above)
Comment (RomanFedytskyi) [reply dev/publicDev/admin.parkingProblem/config.js]: fixed

File: dev/publicDev/admin.properties/config.js (hunk 21 above)
Comment (RomanFedytskyi) [reply dev/publicDev/admin.properties/config.js]: fixed

File: dev/publicDev/admin.properties/config.js (hunk 22 above)
Comment (RomanFedytskyi) [reply dev/publicDev/admin.properties/config.js]: fixed

File: dev/publicDev/scripts/resolver.js (hunk 24 above)
Comment (RomanFedytskyi) [reply dev/publicDev/scripts/resolver.js]: fixed

Comment (FeditskiyRoman) [inline dev/publicDev/scripts/resolver.js]: fixed

File: dev/publicDev/admin.properties/config.js (hunk 22 above)
Comment (FeditskiyRoman) [inline dev/publicDev/admin.properties/config.js]: fixed

File: dev/publicDev/scripts/userCheckingService.js (hunk 17 above)
Comment (RomanFedytskyi) [reply dev/publicDev/scripts/userCheckingService.js]: fixed

File: dev/publicDev/scripts/userCheckingService.js (hunk 19 above)
Comment (RomanFedytskyi) [reply dev/publicDev/scripts/userCheckingService.js]: fixed

File: dev/publicDev/scripts/userCheckingService.js (hunk 18 above)
Comment (RomanFedytskyi) [reply dev/publicDev/scripts/userCheckingService.js]: fixed

**Pull request merged**